samza-core/src/main/java/org/apache/samza/operators/triggers/CountTriggerImpl.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.operators.triggers;
+
+import org.apache.samza.operators.data.MessageEnvelope;
+
+/**
+ * Implementation class for a {@link CountTrigger}
+ */
+public class CountTriggerImpl<M extends MessageEnvelope> extends TriggerImpl<M> {
+
+  private final long triggerCount;
+  private long currentCount;
+  private boolean cancelled = false;
+
+  public CountTriggerImpl(CountTrigger<M> triggerCount, TriggerContext context, TriggerCallbackHandler handler) {
```

Comment:
I see a fair amount of boilerplate around propagating the context and handler. I made some comments on this below. I wonder if we couldn't just save these in one place and pass them into message instead? That would simplify each one of these TriggerImpl subclasses.